### ISSUE TITLE
POI - Osmand not responding (slow)

### ISSUE THREAD

**Issue by Barrocas** [osmandapp/Osmand#2363]
What version of OsmAnd are you using, on what device and Android/iOS version?
2.3.0#16214M, Bq Aquaris E6, Android 4.4.2

Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?
Latest Portugal + Spain Offline maps + live Updates

..

Clicking on a POI on the map, Osmand freezes and lags for some 12s, and Android sistems suggest closing Osmand because it's not responding (slow).

After about 12s the chosen POI is marked. Clicking on another POI freezes again.

**Comment by CJTmmr** [comment]
Same here on Samsung Galaxy Camera; Android 4.3; with nightly's since last Monday March 21st.

Looks like OsmAnd is first completing ReverseGeoLookUp (meanwhile unresponsive), and only then shows ContextMenu.

Previously ContextMenu was shown first, and you could choose next action, while GeoLookUp was running.

Kees-Jan

**Comment by vshcherb** [comment]
Couldn't reproduce it. Is it possible to try latest build or provide more info

**Comment by Barrocas** [comment]
With today night build it is not as slow, but it is still taking about ~3 seconds to respond to a click on an empty spot (to add a POI), or when clicking on an existing POI to modify it.

**Comment by vshcherb** [comment]
I'm using nightly builds all the time.
It is not blazingly fast on my phone it takes 1 second and animation is a bit laggish, so I don't know if it is possible to show the video from your side it will be easier understand what could be a reason.

**Comment by Barrocas** [comment]
I have no way of making video.

But the issue that started the topic is no longer happening.
before it took 12s, and now it's just a small time lag of 1 to 3s
between the click in a spot/POI and any visible response from Osmand.

It used to be faster, now It's acceptable.

-- 
Cumprimentos,
D.er Hugo Barrocas

**Comment by Barrocas** [comment]
Well, it still hangs.

I was navigating, and made an audio note to mark the street name.
Then I clicked on another nearby street, and... Lag... 

![screenshot_2016-03-29-15-11-48](https://cloud.githubusercontent.com/assets/4006691/14111190/afa525d2-f5c1-11e5-8f14-6728b65c1f63.png)

I tried re-installing the night build from day 14, and although it's not what I would call fast, but it responds a lot better than today's build, it doesn't lag as much.
